### PR TITLE
Deprecates SimpleServiceInstance in favor of DefaultServiceInstance

### PR DESCRIPTION
**Pull request by alchemy24**
Please review the code changes for the Issue 824 (https://github.com/spring-cloud/spring-cloud-commons/issues/824)

Unit Tests passed locally via mvn install -P spring

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@alchemy24 Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-commons&pullRequestId=835)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-commons&pullRequestId=835) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by OlgaMaciaszek** [comment]
Fixes gh-824.

**Comment by pivotal-issuemaster** [comment]
@alchemy24 Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-commons&pullRequestId=835)!

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
@alchemy24 Thanks for submitting the PR. I've added some comments for you to address. Also, please add your name to class authors with `@author` tag to all the classes that you've modified.

**Comment by OlgaMaciaszek** [comment]
Fixes gh-824.

File: spring-cloud-commons/src/test/java/org/springframework/cloud/client/loadbalancer/reactive/ReactorLoadBalancerExchangeFilterFunctionTests.java [hunk 1]
```diff
@@ -67,9 +67,11 @@
 
 	@BeforeEach
 	void setUp() {
-		SimpleDiscoveryProperties.SimpleServiceInstance instance = new SimpleDiscoveryProperties.SimpleServiceInstance();
+		DefaultServiceInstance instance = new DefaultServiceInstance(null, null, null, 0,
```
[the remark this answers is not in the document]
Comment (alchemy24) [reply spring-cloud-commons/src/test/java/org/springframework/cloud/client/loadbalancer/reactive/ReactorLoadBalancerExchangeFilterFunctionTests.java:70]: Sure. I will make the change

File: spring-cloud-commons/src/test/java/org/springframework/cloud/client/loadbalancer/reactive/LoadBalancerExchangeFilterFunctionTests.java [hunk 2]
```diff
@@ -69,9 +69,11 @@
 
 	@Before
 	public void before() {
-		SimpleServiceInstance instance = new SimpleServiceInstance();
+		DefaultServiceInstance instance = new DefaultServiceInstance(null, null, null, 0,
```
[the remark this answers is not in the document]
Comment (alchemy24) [reply spring-cloud-commons/src/test/java/org/springframework/cloud/client/loadbalancer/reactive/LoadBalancerExchangeFilterFunctionTests.java:72]: Sure. I will make the change

File: spring-cloud-commons/src/main/java/org/springframework/cloud/client/DefaultServiceInstance.java [hunk 3]
```diff
@@ -29,17 +29,19 @@
...
+	private String serviceId;
 
-	private final String host;
+	private String host;
 
-	private final int port;
+	private int port;
 
-	private final boolean secure;
+	private boolean secure = false;
```
[the remark this answers is not in the document]
Comment (alchemy24) [reply spring-cloud-commons/src/main/java/org/springframework/cloud/client/DefaultServiceInstance.java:40]: Sounds good

File: spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/simple/reactive/SimpleReactiveDiscoveryProperties.java [hunk 4]
```diff
@@ -45,30 +46,31 @@
...
-	private Map<String, List<SimpleServiceInstance>> instances = new HashMap<>();
+	private Map<String, List<DefaultServiceInstance>> instances = new HashMap<>();
 
 	/**
 	 * The properties of the local instance (if it exists). Users should set these
 	 * properties explicitly if they are exporting data (e.g. metrics) that need to be
 	 * identified by the service instance.
 	 */
-	private SimpleServiceInstance local = new SimpleServiceInstance();
+	private DefaultServiceInstance local = new DefaultServiceInstance(null, null, null, 0,
```
[the remark this answers is not in the document]
Comment (alchemy24) [reply spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/simple/reactive/SimpleReactiveDiscoveryProperties.java:56]: Sure. Will make the change

File: spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/simple/SimpleDiscoveryProperties.java [hunk 5]
```diff
@@ -77,15 +79,20 @@ public void setOrder(int order) {
...
 		}
 	}
 
+	public void setInstance(String serviceId, String host, int port) {
+		local = new DefaultServiceInstance(null, serviceId, host, port, false);
+	}
+
 	/**
 	 * Basic implementation of {@link ServiceInstance}.
 	 */
```
[the remark this answers is not in the document]
Comment (alchemy24) [reply spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/simple/SimpleDiscoveryProperties.java:94]: Sounds good. Will make the change

**Comment by alchemy24** [comment]
@OlgaMaciaszek Have fixed all the review comments. Let me know if it looks good

**Review by OlgaMaciaszek — APPROVED** [review]

**Pull request merged**